dhall/src/Dhall/Import.hs

```diff
@@ -1119,3 +1121,52 @@ instance Show ImportResolutionDisabled where
 assertNoImports :: MonadIO io => Expr Src Import -> io (Expr Src Void)
 assertNoImports expression =
     Dhall.Core.throws (traverse (\_ -> Left ImportResolutionDisabled) expression)
+
+{-| This function is used by the @--transitive@ flag of the
+    @dhall {freeze,format,lint}@ subcommands to determine which dependencies
+    to descend into
+-}
+dependencyToFile :: Status -> Import -> IO (Maybe FilePath)
+dependencyToFile status import_ = flip State.evalStateT status $ do
+    parent :| _ <- zoom stack State.get
+
+    child <- fmap chainedImport (hoist liftIO (chainImport parent import_))
```

Comment:
I don't grok what's going on here, but I'm thankful that the tricky bits are so localized! :)

Reply:
Basically, all it's doing is `chainImport`, and the rest is just massaging the type to make it match the surrounding monad transformer stack.  You might find this `mmorph` tutorial helpful:

https://hackage.haskell.org/package/mmorph-1.1.3/docs/Control-Monad-Morph.html#g:3